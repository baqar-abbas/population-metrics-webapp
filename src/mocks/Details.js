import React from 'react';

const Details = () => (
  <>
    <div className="details-container">
      <div className="heading">
        <img src="img" alt="flag" className="flag-image" />
        <h2>
          Country :
          {' '}
          <span>Kenya</span>
        </h2>
      </div>
      <div className="more-details">
        <p>
          <span>Population :</span>
          <span>28875</span>
        </p>
        <p>
          <span>Capital City :</span>
          <span>Nairobi</span>
        </p>
        <p>
          <span> Nation Size :</span>
          <span>500</span>
        </p>
        <p>
          <span> AlphaCode :</span>
          <span>KEN</span>
        </p>
        <p>
          <span> Region :</span>
          <span>Africa</span>
        </p>
        <p>
          <span> Calling Code :</span>
          <span>254</span>
        </p>
      </div>
    </div>
  </>
);

export default Details;
