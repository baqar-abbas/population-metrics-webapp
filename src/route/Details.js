import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { BiArrowToRight } from 'react-icons/bi';

function Details() {
  const params = useParams();
  const nations = useSelector((store) => store.nationReducer);
  const eachNation = nations.find(
    (nation) => nation.name === params.nationName,
  );

  useEffect(() => window.scrollTo(0, 0), []);
  return (
    <>
      <div className="details-container">
        <div className="heading">
          <img
            src={eachNation.flag}
            alt={`${eachNation.name} - flag`}
            className="flag-image"
          />
          <h2>
            Country :
            {' '}
            <span>{eachNation.name}</span>
          </h2>
        </div>
        <div className="more-details">
          <p>
            <span>Population :</span>
            <div className="details-span">
              <span>{eachNation.population}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span>Capital City :</span>
            <div className="details-span">
              <span>{eachNation.capital}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span> Nation Size :</span>
            <div className="details-span">
              <span>{eachNation.area}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span> AlphaCode :</span>
            <div className="details-span">
              <span>{eachNation.id}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span> Region :</span>
            <div className="details-span">
              <span className="details-span2">{eachNation.region}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span> Calling Code :</span>
            <div className="details-span">
              <span>{eachNation.callingcode}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
          <p>
            <span> Time Zone :</span>
            <div className="details-span">
              <span>{eachNation.timezone}</span>
            </div>
            <BiArrowToRight className="toDetails" />
          </p>
        </div>
      </div>
    </>
  );
}

export default Details;
