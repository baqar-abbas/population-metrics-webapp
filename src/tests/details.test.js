import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../mocks/Details';

describe('Companies', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Details />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
