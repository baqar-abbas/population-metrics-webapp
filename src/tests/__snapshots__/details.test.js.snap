// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies renders correctly 1`] = `
<div
  className="details-container"
>
  <div
    className="heading"
  >
    <img
      alt="flag"
      className="flag-image"
      src="img"
    />
    <h2>
      Country :
       
      <span>
        Kenya
      </span>
    </h2>
  </div>
  <div
    className="more-details"
  >
    <p>
      <span>
        Population :
      </span>
      <span>
        28875
      </span>
    </p>
    <p>
      <span>
        Capital City :
      </span>
      <span>
        Nairobi
      </span>
    </p>
    <p>
      <span>
         Nation Size :
      </span>
      <span>
        500
      </span>
    </p>
    <p>
      <span>
         AlphaCode :
      </span>
      <span>
        KEN
      </span>
    </p>
    <p>
      <span>
         Region :
      </span>
      <span>
        Africa
      </span>
    </p>
    <p>
      <span>
         Calling Code :
      </span>
      <span>
        254
      </span>
    </p>
  </div>
</div>
`;
